src/bench_wiring.rs

```diff
@@ -1,54 +1,31 @@
 use criterion::{black_box, Criterion};
-use tokio::runtime::Runtime;
-use wiring::prelude::{Unwire, Unwiring, Wire, Wiring};
-use zstd::zstd_safe::WriteBuf;
-
-pub fn bench<'a, T>(name: &'static str, c: &mut Criterion, data: &'a T)
-where
-    &'a T: Wiring,
-    T: Unwiring + PartialEq,
-{
-    async fn wire_data<W: Wire, T: Wiring>(mut wire: W, data: T) {
-        (&mut wire).wire(data).await.unwrap();
-    }
-
-    let rt = Runtime::new().unwrap();
-    const BUFFER_LEN: usize = 50_000_000;
+use wiring::prelude::{BufUnWire, BufWire, Unwiring, Wiring};
+
+pub fn bench<T: Wiring + Unwiring + PartialEq>(name: &'static str, c: &mut Criterion, data: &T) {
+    const BUFFER_LEN: usize = 10_000_000;
 
     let mut group = c.benchmark_group(format!("{}/wiring", name));
 
-    let mut wire: Vec<u8> = Vec::with_capacity(BUFFER_LEN);
+    let mut wire: Vec<u8> = Vec::new();
+    wire.reserve(BUFFER_LEN); // Optional
 
     group.bench_function("serialize", |b| {
-        b.to_async(Runtime::new().unwrap()).iter_batched(
-            || wire.clone(),
-            |w| wire_data(black_box(w), black_box(data)),
-            criterion::BatchSize::SmallInput,
-        );
-    });
-
-    rt.block_on(async {
-        (&mut wire).wire(data).await.unwrap();
+        b.iter(|| {
+            black_box(BufWire::new(&mut wire).wire(black_box(data)).unwrap());
+        })
     });
 
-    let mut unwire = std::io::Cursor::new(wire);
-
+    let buffer = wire.as_slice();
```

Comment:
Do an extra round of unconditional serialization; we can't assume that `wire` was written to because the user may be filtering their benches (e.g. `cargo bench -- nothing`).

Reply:
Oh I see, sure thing.
Thank you for pointing this